// @tenderly/actions-test/lib/event.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestLog = exports.TestTransactionEvent = exports.TestBlockEvent = exports.TestWebhookEvent = exports.TestPeriodicEvent = void 0;
class TestPeriodicEvent {
    constructor() {
        this.time = new Date(Date.now());
    }
}
exports.TestPeriodicEvent = TestPeriodicEvent;
class TestWebhookEvent {
    constructor(payload) {
        this.payload = payload;
        this.time = new Date(Date.now());
    }
}
exports.TestWebhookEvent = TestWebhookEvent;
class TestBlockEvent {
    constructor() {
        this.blockHash = "0x";
        this.blockNumber = 0;
        this.network = "0";
        this.blockDifficulty = "0";
        this.totalDifficulty = "0";
    }
}
exports.TestBlockEvent = TestBlockEvent;
class TestTransactionEvent {
    constructor() {
        this.blockHash = "0x";
        this.blockNumber = 0;
        this.from = "0x";
        this.hash = "0x";
        this.network = "0";
        this.to = "0x";
        this.logs = [new TestLog()];
        this.input = "0x";
        this.value = "0x";
        this.nonce = "0x";
        this.gas = "0x";
        this.gasUsed = "0x";
        this.cumulativeGasUsed = "0x";
        this.gasPrice = "0x";
        this.gasTipCap = "0x";
        this.gasFeeCap = "0x";
        this.transactionHash = "0x";
    }
}
exports.TestTransactionEvent = TestTransactionEvent;
class TestLog {
    constructor() {
        this.address = "0x";
        this.data = "0x";
        this.topics = ["0x"];
    }
}
exports.TestLog = TestLog;
