// @tenderly/actions-test/lib/storage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestStorage = void 0;
class TestStorage {
    constructor() {
        this.stored = new Map();
    }
    delete(key) {
        this.stored.delete(key);
        return Promise.resolve(undefined);
    }
    getBigInt(key) {
        let value = this.stored.get(key);
        if (value === undefined) {
            return Promise.resolve(BigInt(0));
        }
        return Promise.resolve(BigInt(value));
    }
    getJson(key) {
        let value = this.stored.get(key);
        if (value === undefined) {
            return Promise.resolve({});
        }
        return Promise.resolve(JSON.parse(value));
    }
    getStr(key) {
        let value = this.stored.get(key);
        if (value === undefined) {
            return Promise.resolve("");
        }
        return Promise.resolve(value);
    }
    getNumber(key) {
        let value = this.stored.get(key);
        if (value === undefined) {
            return Promise.resolve(0);
        }
        return Promise.resolve(Number(value));
    }
    putNumber(key, value) {
        this.stored.set(key, value.toString());
        return Promise.resolve();
    }
    putBigInt(key, value) {
        this.stored.set(key, value.toString());
        return Promise.resolve();
    }
    putJson(key, value) {
        this.stored.set(key, JSON.stringify(value));
        return Promise.resolve();
    }
    putStr(key, value) {
        this.stored.set(key, value);
        return Promise.resolve();
    }
    clear() {
        this.stored = new Map();
    }
}
exports.TestStorage = TestStorage;
